pkg/arvo/sur/group-hook.hoon

```diff
@@ -1,12 +1,26 @@
+/-  *group, store=group-store
 |%
-+$  group-hook-action
-  $%  [%add =ship =path]             :: if ship is our, make the group publicly
-                                     :: available for other ships to sync
-                                     :: if ship is foreign, delete any local
-                                     :: group at that path and mirror the
-                                     :: foreign group at our local path
-                                     ::
-      [%remove =path]                :: remove the path.
+::  $action: request to change group-hook state
+::
+::    %add:
+::      if ship is ours make group available to sync, else sync foreign group
```

Comment:
It’s important to keep this pattern around such that work on federation, CRDTs, etc goes smoother when/if we implement it for this store. See Harrison’s chat work for an example of us moving towards those affordances.

Reply:
> It’s important to keep this pattern around

You mean the "explicit add for local" pattern, right?

We're doing a rewrite for correctness and reduction in complexity. If we want to account for, idk federated groups or something, we should probably do so explicitly rather than leave old and otherwise undesirable patterns standing.

Reply:
I'd be concerned about state migration if we split in group-hook in two. Additionally as groups doesn't have a `-view` agent, there are some cross-cutting concerns that complicate a bifurcation (invites etc.).

Reply:
This isn't an undesirable pattern. I will say that I think groups should be split into a `push-hook` and a `pull-hook` now akin to Harrison's work as opposed to waiting to do so until later.

Reply:
Happy to discuss a specific strategy.

Reply:
Is this happening y/n?

Reply:
yeah sounds good